src/Http/Session.php

```diff
@@ -604,15 +604,39 @@ public function renew()
 
         $this->start();
         $params = session_get_cookie_params();
-        setcookie(
-            session_name(),
-            '',
-            time() - 42000,
-            $params['path'],
-            $params['domain'],
-            $params['secure'],
-            $params['httponly']
-        );
+        if (PHP_VERSION_ID >= 70300) {
+            // PHP 7.3 and up don't support adding SameSite to the path,
+            // but require the alternative setcookie syntax.
+            // So we extract sameSite from the path
+            $sameSite = null;
+            $path = $params['path'];
+            if (preg_match('/^(.*); SameSite=(.*)$/', $path, $matches) === 1) {
+                $path = $matches[1];
+                $sameSite = $matches[2];
+            }
+            setcookie(
+                session_name(),
+                '',
+                [
+                    'expires' => time() - 42000,
+                    'path' => $path,
+                    'domain' => $params['domain'],
+                    'secure' => $params['secure'],
+                    'httponly' => $params['httponly'],
+                    'samesite' => $sameSite,
```

Comment:
Is this actually necessary for "deleting" the cookie?

I'm not sure if I remember this correctly, but isn't only the path and the domain required to match for a given cookie name in order to be able to update the cookie?

Reply:
Ah that I don't know. I could remove this here indeed if we don't need the samesite. But do we then need the other params?

Reply:
I'm not sure about that, hence why I'm asking ;)

The PHP docs say:

> Cookies must be deleted with the same parameters as they were set with. If the value argument is an empty string, or FALSE, and all other arguments match a previous call to setcookie, then the cookie with the specified name will be deleted from the remote client. This is internally achieved by setting value to 'deleted' and expiration time to one year in past. 

But from looking at the source this is already not quite correct, the expiration date is different, and it only depends on the value, not on any other arguments: https://github.com/php/php-src/blob/php-7.4.10/ext/standard/head.c#L110-L122

From a quick test it seems that it's not required to specify all the same attributes, [**which is also what the RFC suggests**](https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05#section-3.1), I can freely change `HttpOnly`, `Secure` and `SameSite`, PHP will generate the same `deleted` cookie value and `Thu, 01-Jan-1970 00:00:01 GMT` expiration value, and the browser will delete the cookie just fine (except of course for combinations like `SameSite=None` without `Secure` which would be rejected by the browser).

But who knows, maybe there are actually cases that require the same exact attributes.

Reply:
I'd say it never hurts to have all attributes? Then it might be better to leave it in.